src/MultiBridgeMessageReceiver.sol

```diff
@@ -73,17 +73,27 @@ contract MultiBridgeMessageReceiver is IMultiBridgeMessageReceiver, ExecutorAwar
     ////////////////////////////////////////////////////////////////*/
 
     /// @notice sets the initial parameters
-    constructor(uint256 _srcChainId, address _gac) {
-        if (_gac == address(0)) {
-            revert Error.ZERO_ADDRESS_INPUT();
-        }
-
+    constructor(uint256 _srcChainId, address _gac, address[] memory _receiverAdapters, uint64 _quorum) {
         if (_srcChainId == 0) {
             revert Error.INVALID_SENDER_CHAIN_ID();
         }
+        if (_gac == address(0)) {
+            revert Error.ZERO_ADDRESS_INPUT();
+        }
 
-        gac = IGAC(_gac);
         srcChainId = _srcChainId;
+        gac = IGAC(_gac);
+
+        uint256 numAdapters = _receiverAdapters.length;
+        bool[] memory operations = new bool[](numAdapters);
+        for (uint256 i; i < numAdapters;) {
+            operations[i] = true;
+            unchecked {
+                ++i;
+            }
+        }
```

Comment:
Suggest we refactor this a bit for efficiency as follows:
- move zero address check in `_updateReceiverAdapters()` lines 270-272 into `_updateReceiverAdapter()`
- iterate through the receiver adapters and call `_updateReceiverAdapter(..,true)`

Reply:
Done.

Reply:
Tests seem to be failing.

Reply:
Looks like it's just flakiness with endpoints.